src/dmd/backend/machobj.d

```diff
@@ -248,6 +248,14 @@ int seg_tlsseg_bss = UNKNOWN;
  * with an initializer.
  */
 int seg_tlsseg_data = UNKNOWN;
+
+int seg_cstring = UNKNOWN;        // __cstring section
+int seg_mod_init_func = UNKNOWN;  // __mod_init_func section
+int seg_mod_term_func = UNKNOWN;  // __mod_term_func section
+int seg_deh_eh = UNKNOWN;         // __deh_eh section
+int seg_textcoal_nt = UNKNOWN;
+int seg_tlscoal_nt = UNKNOWN;
+int seg_datacoal_nt = UNKNOWN;
```

Comment:
If any of these can't be negative, they ought to be `uint`. You only need one test for bounds checking etc.

Reply:
UNKNOWN is -1. The OMF generation also uses negative seg numbers.